### ISSUE TITLE
hosting_site_backup_manager tasks not running after 2.4.9 -> 3.0.0 upgrade

### ISSUE THREAD

**Issue by cvining** [omega8cc/boa#901]
Since running the barracuda & octopus upgrades to 3.0.0 I noticed the automatic backup tasks were not getting executed.  Always possible I messed up the upgrade process of course, but AFAIK I followed the usual instructions. 

In my debugging, I ran the cron job for runner.sh as root, which reported a function 'hosting_get_most_recent_task()' not found error in the hosting_backup_queue module. Here's the output

```
_do:~# /usr/bin/nice -n19 /usr/bin/ionice -c2 -n7 bash /var/xdrago/runner.sh         
load is 26 while maxload is 600
running /var/xdrago/run-o1
waiting 4 sec
Drush command terminated abnormally due to an unrecoverable error.   [error]
Error: Call to undefined function hosting_get_most_recent_task() in
/data/disk/o1/aegir/distro/008/profiles/hostmaster/modules/aegir/hosting_site_backup_manager/hosting_backup_queue/hosting_backup_queue.module, line 71_
```

And, indeed, there does not appear to be a function named 'hosting_get_most_recent_task()' anywhere in the current version.

I hacked the hosting_backup_queue.module file to remove the offending call and runner.sh is executing tasks as expected now, but I don't have confidence it's a proper fix.

Hopefully it's as simple as using a more recent version of hosting_site_backup_manager

Thanks for your attention, and all your hard work. It's all much appreciated.

Here is my 'boa info more' details:     https://gist.github.com/cvining/c20d93533ec0d38f1664d7c0a5f8cd43#file-boa-info-20160410

**Comment by omega8cc** [comment]
Thanks for the report! We don't use this module and instead recommend to use the `backboa` tool, so maybe we overlooked something, but I though that we have bundled latest version?

**Comment by omega8cc** [comment]
@cvining Please report this at https://www.drupal.org/project/issues/hosting_site_backup_manager 

BOA comes with the most recent code and indeed it calls the not existing, 6.x specific `hosting_get_most_recent_task()`

**Comment by omega8cc** [comment]
It seems that it is a known issue: https://www.drupal.org/node/1063674#comment-11005589

**Comment by cvining** [comment]
Thanks for the responses! I've posted a comment on the hosting_site_backup_manager issues page confirming the known bug. And I'll take this as an opportunity to look into the backboa tool.

I'm not certain of the etiquette here. Should I close this issue since it's been added as a 3.0.2 milestone? Or leave it open until a fix is actually in place?

**Comment by omega8cc** [comment]
@cvining Let's leave this open. The milestone is for now only a placeholder to keep an eye on this until the problem is fixed in the contrib. Thank you!

**Comment by EdNett** [comment]
AEgir has a May 19 update that will probably take care of this problem; I have the same problem on wheezy with php56-fpm

**Comment by omega8cc** [comment]
There are no fixes in this module since March 28, so the latest Aegir release doesn't improve anything.

**Comment by florisvanwoudenberg** [comment]
I did use the backup manager and was not aware of back-boa yet. Thanks again, I will test back-boa to see if this suits my needs.

**Comment by florisvanwoudenberg** [comment]
I have on question: I installed backboa and ran the first backup. But I do not see any database backups. How do I have daily or hourly database backups?

**Comment by omega8cc** [comment]
@florisvanwoudenberg Please check:

https://github.com/omega8cc/boa/blob/master/README.txt#L138
https://github.com/omega8cc/boa/blob/master/docs/BACKUPS.txt#L8

**Comment by florisvanwoudenberg** [comment]
Cool, thanks!

**Comment by tourtools** [comment]
helmo's [patch](http://cgit.drupalcode.org/hosting/commit/task/hosting_task.module?id=313c4e7226d9c0e5b86f2713ead8cec7554a14d8) solves the issue

**Comment by omega8cc** [comment]
@tourtools Where from the second patch is? (the issue URL?)

**Comment by cvining** [comment]
This issue seems to be resolved by upgrading to the latest 3.1.2. Thx! For me, the issue can be closed now.

**Comment by omega8cc** [comment]
Thanks!
